test/integration/gateway_webhook_test.go

```diff
@@ -0,0 +1,165 @@
+package integration
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/annotations"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/controllers/gateway"
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters/types/kind"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	admregv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+	gatewayclient "sigs.k8s.io/gateway-api/pkg/client/clientset/gateway/versioned"
+)
+
+func TestGatewayValidationWebhook(t *testing.T) {
+	t.Parallel()
+	if env.Cluster().Type() != kind.KindClusterType {
+		t.Skip("TODO: webhook tests are only supported on KIND based environments right now")
```

Comment:
I know this has been the historical language but I feel like it's just a good a time as any to clean it up a bit:

```suggestion
		t.Skip("webhook tests are only available on kind clusters currently")
```

Reply:
da139293